Hacktoberfest: Migrate docs from confluence (#11)
jenkinsci/publish-over-plugin: README.md

#### Port

The port that the server is running on can be changed if the server that you are connecting to is not running on the default port for the service that the plugin talks to.

#### Timeout (ms)

The connect timeout can be configured (in milliseconds)

### Publisher

There are some options that are set at the publisher level - options that affect all of the server connections within.

All of the publisher options are only exposed when you click on the "Advanced" button (at the bottom of the configuration for this plugin).

#### Advanced options

##### Publish to other servers if an error occurs

By default, when an error occurs during publishing the publisher will fail and return immediately.

Set this option to cause the publisher to try to publish to other servers after publishing to a previous configuration has failed.

##### Fail the build if an error occurs

By default, when an error occurs, the publisher will set the build result to UNSTABLE.

Setting this option will cause a failure in the publisher to set the build result to FAILED.

This option is especially useful in the case of a promotion where the main action is to Publish Over ...

> ##### Always send from master

Reviewer: We've deprecated the term "master" in favor of "controller"
```suggestion
##### Always send from controller
```

> By default, the publisher will connect from the host that has the files that need to be transferred - if a build is performed on a slave, then the transfer would be initiated from the slave.

Reviewer: We've switched from "slave" to "agent"
```suggestion
By default, the publisher will connect from the host that has the files that need to be transferred - if a build is performed on an agent, then the transfer would be initiated from the agent.
```

> By selecting this option, the connection will be initiated from the Jenkins master.

Reviewer: ```suggestion
By selecting this option, the connection will be initiated from the Jenkins controller.
```

May be useful for those with exciting network configurations/administrators.

> ##### Give the master a NODE_NAME

Reviewer: ```suggestion
##### Give the controller a NODE_NAME
```

Legacy option

In Jenkins 1.414, the Jenkins master will be assigned a NODE_NAME ('master')

For the reason above, this option will not appear when this plugin is installed on Jenkins 1.414 and later.

Additionally, the option now defaults to 'master' when this plugin is installed on a Jenkins older than 1.414.

Sets NODE_NAME to the value specified if the environment variables contain a variable NODE_NAME that does not have a value.

> If you are running on multiple nodes and you have executors on the master Jenkins and the build may occur on the master, then if you set this option, the NODE_NAME for the master will be set to the configured value.

Reviewer: ```suggestion
If you are running on multiple nodes and you have executors on the Jenkins controller and the build may occur on the controller, then if you set this option, the NODE_NAME for the controller will be set to the configured value.
```
